src/codewhisperer/util/telemetryHelper.ts

```diff
@@ -317,43 +317,44 @@ export class TelemetryHelper {
         if (e2eLatency < 0) {
             e2eLatency = performance.now() - session.invokeSuggestionStartTime
         }
-
-        client
-            .sendTelemetryEvent({
-                telemetryEvent: {
-                    userTriggerDecisionEvent: {
-                        sessionId: sessionId,
-                        requestId: this.sessionDecisions[0].codewhispererFirstRequestId,
-                        customizationArn: selectedCustomization.arn === '' ? undefined : selectedCustomization.arn,
-                        programmingLanguage: {
-                            languageName: runtimeLanguageContext.toRuntimeLanguage(
-                                this.sessionDecisions[0].codewhispererLanguage
-                            ),
+        if (this.sessionDecisions[0].codewhispererSessionId) {
+            client
+                .sendTelemetryEvent({
+                    telemetryEvent: {
+                        userTriggerDecisionEvent: {
+                            sessionId: this.sessionDecisions[0].codewhispererSessionId,
```

Comment:
The `sessionId` function parameter is no longer used, should it be removed?

Reply:
have a high level q: 
why passing `this.seesionDecisions[0].seesionId` could solve the issue?
IMO, passing here the function argument of `sendUserTriggerDecisionTelemetry(sessionId: string)` would be more robust and if it doesn't work correctly why using a "state" variable could fix this issue.

Reply:
would it help if we simply make `sendUserTriggerDecisionTelemetry` an async call and await on the call sites instead of running async

Reply:
I think the issue here is when next session calls sendUserTriggerDecisionTelemetry, it needs to send the previous session, not the sessionID from current session?

Reply:
I see, that make sense

Reply:
@leigaol  we should add a comment to note down this

Reply:
We should always use sessionId, let's make it to be the correct value but not abandon it. The logic should be the events for the old sessionId should be sent before it gets assigned to a new one.

Reply:
I would recommend to discard the usage of this.sessionDecisions, it feels better to manage a single sessionId per session in the `codewhispererSession.ts`, let's discuss on Monday on how to fix this.